hyperspy/_signals/signal2d.py

```diff
@@ -605,3 +608,433 @@ def add_ramp(self, ramp_x, ramp_y, offset=0):
+        """
+        arr_shape = (self.axes_manager._navigation_shape_in_array
+                     if self.axes_manager.navigation_size > 0
+                     else [1, ])
+        peaks = np.zeros(arr_shape, dtype=object)
+        for z, indices in zip(self._iterate_signal(),
+                              self.axes_manager._array_indices_generator()):
+            if method == 'skimage':
+                peaks[indices] = peak_local_max(z, *args, **kwargs)
+            elif method == 'max':
+                peaks[indices] = find_peaks_max(z, **kwargs)
+            elif method == 'minmax':
+                peaks[indices] = find_peaks_minmax(z, **kwargs)
+            elif method == 'zaefferer':
+                peaks[indices] = find_peaks_zaefferer(z, **kwargs)
+            elif method == 'stat':
+                peaks[indices] = find_peaks_stat(z, **kwargs)
+            elif method == 'laplacian_of_gaussians':
+                peaks[indices] = find_peaks_log(z, **kwargs)
+            elif method == 'difference_of_gaussians':
+                peaks[indices] = find_peaks_dog(z, **kwargs)
+            else:
+                raise NotImplementedError("The method `{}` is not implemented. "
+                                          "See documentation for available "
+                                          "implementations.".format(method))
+
+        return peaks
+
+    def find_peaks2D_interactive(self):
+        from hyperspy.gui import peakfinder2D
+        """
+        Find peaks using an interactive tool.
+
+        Notes
+        -----
+        Requires `ipywidgets` and `traitlets` to be installed.
+
+        """
+        peakfinder = peakfinder2D.PeakFinderUIIPYW()
+        peakfinder.interactive(self)
+
+
+NO_PEAKS = np.array([[np.nan, np.nan]])
+
+
+def clean_peaks(peaks):
+    if len(peaks) == 0:
+        return NO_PEAKS
+    else:
+        return peaks
+
+
+def find_peaks_minmax(z, separation=5., threshold=10., interpolation_order=3):
+    """
+    Method to locate the positive peaks in an image by comparing maximum
+    and minimum filtered images.
+    Parameters
+    ----------
+    z: ndarray
+    separation: expected distance between peaks
+    threshold: ???
+    Returns
+    -------
+    peaks: array with dimensions (npeaks, 2) that contains the x, y coordinates
+           for each peak found in the image.
+    """
+    data_max = ndi.filters.maximum_filter(z, separation)
+    maxima = (z == data_max)
+    data_min = ndi.filters.minimum_filter(z, separation)
+    diff = ((data_max - data_min) > threshold)
+    maxima[diff == 0] = 0
+    labeled, num_objects = ndi.label(maxima)
+    peaks = np.array(
+        ndi.center_of_mass(z, labeled, range(1, num_objects + 1)))
+
+    return clean_peaks(peaks)
+
+
+def find_peaks_max(z, alpha=3., size=10):
+    """
+    Method to locate positive peaks in an image by simple local maximum
+    searching.
+    """
+    # preallocate lots of peak storage
+    k_arr = np.zeros((10000, 2))
```

Comment:
No need to use a numpy array. Just initialize as an empty list and append when found, create the array at the end.